virtcontainers/container.go

```diff
@@ -418,61 +418,64 @@ func newContainer(pod *Pod, contConfig ContainerConfig) (*Container, error) {
 
 // createContainer creates and start a container inside a Pod. It has to be
 // called only when a new container, not known by the pod, has to be created.
-func createContainer(pod *Pod, contConfig ContainerConfig) (*Container, error) {
+func createContainer(pod *Pod, contConfig ContainerConfig) (c *Container, err error) {
 	if pod == nil {
 		return nil, errNeedPod
 	}
 
-	c, err := newContainer(pod, contConfig)
+	c, err = newContainer(pod, contConfig)
 	if err != nil {
-		return nil, err
+		return
 	}
 
-	if err := c.createContainersDirs(); err != nil {
-		return nil, err
+	if err = c.createContainersDirs(); err != nil {
+		return
 	}
 
-	if !c.pod.config.HypervisorConfig.DisableBlockDeviceUse {
-		agentCaps := c.pod.agent.capabilities()
-		hypervisorCaps := c.pod.hypervisor.capabilities()
-
-		if agentCaps.isBlockDeviceSupported() && hypervisorCaps.isBlockDeviceHotplugSupported() {
-			if err := c.hotplugDrive(); err != nil {
-				return nil, err
-			}
+	// In case the container creation fails, the following takes care
+	// of rolling back all the hotplug and mount operations.
+	defer func() {
+		if err != nil && c != nil {
+			c.removeResources()
```

Comment:
Similar to https://github.com/kata-containers/agent/pull/196, could you add some log calls if these "undo" calls fail?

Reply:
Yes, working on it !